src/components/views/EditUserPreferencePage.js

```diff
@@ -1,13 +1,209 @@
+        if (!Array.isArray(newValue)) {
+            return;
+        }
+
+        if (activeThumb === 0) {
+            setDistance([Math.min(newValue[0], distance[1]), distance[1]]);
+            console.log(distance)
+        } else {
+            setDistance([distance[0], Math.max(newValue[1], distance[0])]);
+            console.log(distance)
+        }
+    };
+
+    //Method to handle changes for the checkboxes gender
+    const handleChangeGender = (event) => {
+        setGenderPreference({
+            ...genderPreference,
+            [event.target.name]: event.target.checked,
+        });
+    };
+
+
+    let editPreferences = (
+        <div className="login form">
+            <h2 className="login title"> Set Preferences </h2>
+            <div className="login container-title">Edit Age Preferences:</div>
+            <Slider
+                value={agePreference}
+                onChange={handleChangeAge}
+                valueLabelDisplay="auto"
+                min={18}
+                max={99}
+                disableSwap
+            />
+                Mininum Age: {agePreference[0]} - Maximum Age: {agePreference[1]}
+            <div className="login container-title">Edit Gender Preferences:</div>
+
+            <FormControl
+                required
+                error={noGenderSelectedError}
+            >
+                <FormLabel component="legend">Pick at least 1</FormLabel>
+                <FormGroup>
+                    <FormControlLabel
+                        control={
+                            <Checkbox checked={MALE} onChange={handleChangeGender} name="MALE" />
+                        }
+                        label="Male"
+                    />
+                    <FormControlLabel
+                        control={
+                            <Checkbox checked={FEMALE} onChange={handleChangeGender} name="FEMALE" />
+                        }
+                        label="Female"
+                    />
+                    <FormControlLabel
+                        control={
+                            <Checkbox checked={OTHER} onChange={handleChangeGender} name="OTHER" />
+                        }
+                        label="Other"
+                    />
+                </FormGroup>
+                <FormHelperText>Select at Least 1 Preference</FormHelperText>
+            </FormControl>
+
+            <div className="login container-title">Edit Distance</div>
+            <Slider //TODO: Maybe add logarithmic scaling to the rangeslider
+                value={distance}
+                onChange={handleChangeDistance}
+                valueLabelDisplay="auto"
+                min={0}
+                max={20000}
+                disableSwap
+            />
+            {distance[0]} Km - {distance[1]} Km
+            <div className= "userPage fixed-button-container">
+                <div className= "userPage moving-button-container">
```

Comment:
A moving button container inside a fixed button container? Seems a bit weird to me, but as long as it works 😄